.github/workflows/generator_generic_slsa3.yml

```diff
@@ -281,3 +330,12 @@ jobs:
           set -euo pipefail
           echo "outcome=$([ "$SUCCESS" == "true" ] && echo "success" || echo "failure")" >> "$GITHUB_OUTPUT"
           [ "$CONTINUE" == "true" ] || [ "$SUCCESS" == "true" ] || exit 27
+
+  # cleanup deletes internal artifacts used by the generator workflow
+  cleanup:
+    runs-on: ubuntu-latest
+    needs: [detect-env, generator, upload-assets]
+    steps:
+      - uses: geekyeggo/delete-artifact@54ab544f12cdb7b71613a16a2b5a37a9ade990af # v2.0.0
```

Comment:
Is it really necessary to delete the artifact? I'm worried a bit about introducing another 3rd party dependency.

Reply:
+1, created https://github.com/slsa-framework/slsa-github-generator/issues/2382 to delete without the Action.
Note that this Action is already used in delegators. It's has 2 releases since 2020 so it won't incur additional dependabot PRs. But folks who use the Action allow-list settings won't like this.